src/cdk-experimental/menu/menu-bar.ts

```diff
@@ -28,10 +44,143 @@ import {CDK_MENU, Menu} from './menu-interface';
     {provide: CDK_MENU, useExisting: CdkMenuBar},
   ],
 })
-export class CdkMenuBar extends CdkMenuGroup implements Menu {
+export class CdkMenuBar extends CdkMenuGroup implements Menu, AfterContentInit, OnDestroy {
   /**
    * Sets the aria-orientation attribute and determines where menus will be opened.
    * Does not affect styling/layout.
    */
   @Input('cdkMenuBarOrientation') orientation: 'horizontal' | 'vertical' = 'horizontal';
+
+  /** All child MenuItem elements nested in this MenuBar. */
+  @ContentChildren(CdkMenuItem, {descendants: true})
+  private readonly _allItems: QueryList<CdkMenuItem>;
+
+  /** Event emitter used to end the subscription to document click events. */
+  private readonly _destroyDocumentSubscription: EventEmitter<void> = new EventEmitter();
+
+  /** A reference to the document. */
+  private readonly _document: Document;
+
+  /** Whether the element is inside of a ShadowRoot component. */
```

Comment:
Here, could you explain the significance of this? What do you differently inside vs outside a ShadowRoot component?

Reply:
I thought it made more sense to place the explanatory comment where it's used https://github.com/angular/components/pull/19774/files#diff-6e90f66886311d8bf2422493cf7c1b84R138-R139